src/sdk.ts

```diff
@@ -740,6 +740,7 @@ export default class CegaEthSDK {
     const { results }: ContractCallResults = await multicall.call(contractCallContext);
 
     const metadatasByVaultAddress = results.vaultMetadataList.callsReturnContext
+      .filter((x) => x.returnValues.length > 0)
```

Comment:
just add a comment:
```
filter out multicalls that return an empty array of vaultMetadatas
```

Also, can you add the same filter to the `getAllVaultMetadataFcn`